### ISSUE TITLE
Incorrect Syntax Hightlighting

### ISSUE THREAD

**Issue by bogdan0083** [thieman/better-search#6]
Hello! Thanks for such an amazing plugin! Really enjoy using it! ❤️

I'm working on JS project and some of language syntax highlighting are not correct:
![image](https://api.monosnap.com/rpc/file/download?id=9A6sK5zfFQocZ7LISmQBdXSAluZOcr)

You can see that document headed renders partly in red, and code is also mostly red. 

I've been checking out source code and found where the problem is:

```typescript
  async provideDocumentLinks(
    document: TextDocument,
    token: CancellationToken
  ): Promise<DocumentLink[]> {
    // Hack, not sure where the best place is to run this. Can't run it
    // in the content provider since I don't have an actual TextDocument yet
    const language = this._languages[document.uri.toString()];
    if (language !== undefined) {
     // right here!
      await languages.setTextDocumentLanguage(document, language as string);
    }
    this._readyToDispose[document.uri.toString()] = true;
    // End hack

``` 

`setTextDocumentLanguage` method sets document language and syntax highlight, but it adds syntax highlighting for the whole document instead of just adding it for search result. That's what causes highlighting to be inconsistent across search.

Would be cool if we find a way to apply syntax highlighting only to search results in scope, ignoring surrounding results.

This issue is not critical, but still it'd be nice to have that fixed someday.

Thanks!

**Comment by thieman** [comment]
You're correct, we have to apply syntax highlighting to the entire document right now as far as I know. This introduces quite a few problems, particularly with block comments that aren't terminated because only one end of it shows up in search results.

> Would be cool if we find a way to apply syntax highlighting only to search results in scope, ignoring surrounding results.

This is exactly the fix but I'm not aware of any way currently to selectively apply syntax highlighting. If VSCode ships something (or if I'm just missing it) I'd be happy to integrate it into the extension.

**Comment by beinginorbit** [comment]
Should I expect it to at least auto detect? Just curious, I primarily work in .js so I can probably default it to that.

**Comment by beinginorbit** [comment]
Those interested in at least defaulting it, config looks something like this in `settings.json`
```  
"files.associations": {
    "Σ*": "javascript"
  },
```

**Comment by thieman** [comment]
@spacesuitdiver I would expect it to autodetect JS extensions based on the handful of extensions I hardcoded https://github.com/thieman/better-search/blob/714ba660abf28a0d2c327c5dfb710039db1d0135/src/providers.ts#L22-L33

What are you seeing it do? No highlighting at all?

**Comment by beinginorbit** [comment]
@thieman it doesn't seem to work this way for some reason on my machine? Is it maybe because I don't have the latest from GitHub? Is it basically creating a search "file" that has that extension, because my extension is just the lambda and keyword, no extension.

**Comment by thieman** [comment]
It's attempting to be clever and picks a single language to highlight the entire search result buffer in based on the **most common** extension in the files returned. If you do a search that only returns a single JS file, does it still not highlight?

**Comment by beinginorbit** [comment]
@thieman no highlighting even when only a single file is found .js extension. I'm using quick search for what it's worth and I don't have a project workspace (just an open folder).

**Comment by beinginorbit** [comment]
Side note, queries with special chars don't seem to work, like `<Div css={{ flex: 1, overflow: 'hidden'`, I can make a separate issue for that.

**Comment by thieman** [comment]
>  I can make a separate issue for that.

Yes, please! It might be a shell escaping issue so please note your platform as well

I'll play around and see if I can repro the highlighting issue. As long as you're on a reasonably recent release it should be working, you don't need to have the latest commit off GitHub or anything.
